src/main.rs

```diff
@@ -99,16 +99,16 @@ fn main() -> Result<(), Box<std::error::Error>> {
     let args = parse_args(std::env::args())?;
     let mapper = match args.flex_file {
         None => Mapper::default(),
-        Some(flex_file) => load_mapper(flex_file)?,
+        Some(flex_file) => load_mapper(&flex_file)?,
     };
 
     let input =
-        if args.words.len() == 0 {
+        if args.words.is_empty() {
```

Comment:
Is there a specific motivation for this? I know that with linked lists this is an optimization, but I wouldn't have thought it would be a performance issue for vectors.